File: today/start.py
import sys
from pathlib import Path
import subprocess

from today.cli import build_parser, parse_args, parse_task_files


def run(args) -> None:
    parser = build_parser()
    cli_args = parse_args(parser, args)
    task_file = Path("/tmp/task")

    if cli_args.task_id is None:
        task_file.write_text("")
    else:
        tasks = parse_task_files(cli_args)
        if cli_args.task_id >= len(tasks):
            print(f"The task id provided ({cli_args.task_id}) is not in range, rerun today")
            sys.exit(1)
        task = tasks[cli_args.task_id]
        path = " → ".join(task.path)
        current_task = f"<span weight='bold'> Current Task ({cli_args.task_id}) -</span>" if False else ""
        rel_path = task.file_path.relative_to(cli_args.task_dir)
        task_snippet = f"<span color='white'>{current_task} {path} <span weight='bold'>→</span> {task.title} ({rel_path})</span>"
        Path("/tmp/task").write_text(task_snippet)

    # https://i3wm.org/docs/i3status.html
    subprocess.run("killall -USR1 i3status", shell=True)
    sys.exit(0)


def main():
    sys.exit(run(sys.argv[1:]))

File: src/bin/start.rs
use clap::Parser;
use today::cli::{CliArgs, parse_task_files};
use std::fs;
use std::path::Path;
use std::process::{self, Command};

fn main() {
    if let Err(e) = run() {
        eprintln!("Error: {}", e);
        process::exit(1);
    }
}

fn run() -> Result<(), String> {
    let args = CliArgs::parse();
    let task_file = Path::new("/tmp/task");

    if args.task_id.is_none() {
        // Clear the task file
        fs::write(task_file, "")
            .map_err(|e| format!("Failed to write to /tmp/task: {}", e))?;
    } else {
        // Parse tasks and write the selected task
        let tasks = parse_task_files(&args)?;
        let task_dir = args.task_dir()?;

        let task_snippet = if let Some(task_id_result) = args.get_task_id() {
            match task_id_result {
                Ok(task_id) => {
                    if task_id >= tasks.len() {
                        return Err(format!(
                            "The task id provided ({}) is not in range, rerun today",
                            task_id
                        ));
                    }
                    let task = &tasks[task_id];
                    let path = task.path.join(" <span weight='bold'>/</span> ");
                    let rel_path = task.file_path.strip_prefix(&task_dir)
                        .unwrap_or(&task.file_path);
                    
                    format!(
                        "<span color='white'> {} <span weight='bold' color='red'>→</span> {} <span color='lightgray'>({}: {})</span></span>",
                        path,
                        task.title,
                        rel_path.display(),
                        task.line_number
                    )
                }
                Err(task_str) => {
                    // Ad-hoc task
                    format!(
                        "<span color='white' weight='bold'>Ad-hoc task:</span> <span color='lightgrey'>{}</span>",
                        task_str
                    )
                }
            }
        } else {
            String::new()
        };

        fs::write(task_file, task_snippet)
            .map_err(|e| format!("Failed to write to /tmp/task: {}", e))?;
    }

    // Signal i3status to reload
    let _ = Command::new("killall")
        .args(["-USR1", "i3status"])
        .output();

    Ok(())
}
